### PR TITLE
mcux-sdk: Update to SDK 2.15.000

### PR DESCRIPTION
**Pull request by dleach02**
Updating NXP HAL to SDK 2.15.000

### CODE REVIEW

**Review by gmarull — DISMISSED** [review]
This PR adds yet more binary blobs (*.FLM), which makes things worse than they already are. Please, remove all the unnecessary stuff (which is a lot). This repo is cloned by everybody following the GSG, and so should be kept in decent shape.

File: mcux/README [hunk 1]
```diff
@@ -4,13 +4,15 @@ MCUXpresso SDK
...
    https://github.com/NXPmicro/mcux-sdk
+   SHA: 076dc8823ac7c8c5499688418aa7691160a10ec6
 
    NXP Kinetis Connectivity Software
    https://www.nxp.com/webapp/Download?colCode=KW40Z-CONNECTIVITY-SOFTWARE&appType=license&Parent_nodeId=1432854896956716810497&Parent_pageType=product
    https://www.nxp.com/webapp/Download?colCode=MKW2XD-MRC20A-CONNECTIVITY-SW&appType=license&Parent_nodeId=1425322332576706617013&Parent_pageType=productI
 
    NXP MCUXpresso SDK USB
    https://github.com/NXPmicro/mcux-sdk-middleware-usb
+   SHA: c8317a37546b18bef2072d6075e061d668b3f1c7
```
Comment (brycewilkins) [inline mcux/README:15]: I don't think [mcux-sdk-middleware-usb](https://github.com/nxp-mcuxpresso/mcux-sdk-middleware-usb) `main` has been updated with commits from the release associated with MCUX SDK v2.15.000.

Do you need to use [this](https://github.com/nxp-mcuxpresso/mcux-sdk-middleware-usb/releases/tag/MCUX_2.15.000) tagged release instead? (Though then you'll miss a recent fix in `main` :/)

Comment (dleach02) [reply mcux/README:15]: The NXP SDK team is starting to shift to a release to main approach. This sha is supposed to be 2.15.000+updates to support the RW61x SOCs

Comment (dleach02) [reply mcux/README:15]: @brycewilkins, a separate HAL PR was merged that included the middleware-usb

https://github.com/zephyrproject-rtos/hal_nxp/pull/316

**Comment by dleach02** [comment]
@gmarull, I've removed the binary files and additional items (hard to review though based on the number of files). Good catch.

File: mcux/README [hunk 2]
```diff
@@ -4,13 +4,15 @@ MCUXpresso SDK
 Origin:
    NXP MCUXpresso SDK
    https://github.com/NXPmicro/mcux-sdk
```
Comment (brycewilkins) [inline mcux/README:6]: The new URL is https://github.com/nxp-mcuxpresso/mcux-sdk

File: mcux/README [hunk 3]
```diff
@@ -4,13 +4,15 @@ MCUXpresso SDK
...
+   SHA: 076dc8823ac7c8c5499688418aa7691160a10ec6
 
    NXP Kinetis Connectivity Software
    https://www.nxp.com/webapp/Download?colCode=KW40Z-CONNECTIVITY-SOFTWARE&appType=license&Parent_nodeId=1432854896956716810497&Parent_pageType=product
    https://www.nxp.com/webapp/Download?colCode=MKW2XD-MRC20A-CONNECTIVITY-SW&appType=license&Parent_nodeId=1425322332576706617013&Parent_pageType=productI
 
    NXP MCUXpresso SDK USB
    https://github.com/NXPmicro/mcux-sdk-middleware-usb
```
Comment (brycewilkins) [inline mcux/README:14]: The new URL is https://github.com/nxp-mcuxpresso/mcux-sdk-middleware-usb

File: mcux/README [hunk 4]
```diff
@@ -4,13 +4,15 @@ MCUXpresso SDK
 Origin:
    NXP MCUXpresso SDK
    https://github.com/NXPmicro/mcux-sdk
+   SHA: 076dc8823ac7c8c5499688418aa7691160a10ec6
```
Comment (brycewilkins) [inline mcux/README:7]: This commit does not include the LTO fix merged to upstream https://github.com/nxp-mcuxpresso/mcux-sdk/pull/166

**Review by brycewilkins — COMMENTED** [review]
Couple of observations from the `README`. 
- A more recent commit from MCUX SDK would be good to grab the LTO fix.

**Comment by dleach02** [comment]
@brycewilkins, thank you for the review and comments. I will be updating this PR soon with the the updates on mcux-sdk after a quick sync with the SDK team.

File: mcux/README (hunk 4 above)
Comment (mmahadevan108) [reply mcux/README:7]: I have pulled in the latest MCUX-SDK release and this commit is included.

Comment (brycewilkins) [reply mcux/README:7]: @mmahadevan108 Can you update the SHA that's listed? It references a much earlier commit **without** the LTO fix.

Comment (mmahadevan108) [reply mcux/README]: Thanks for pointing this, I have updated the SHA

Comment (brycewilkins) [reply mcux/README]: @mmahadevan108 Thank you! Having the SHA's listed in the `README` is a great addition -- super helpful!

**Review by hakehuang — APPROVED** [review]
full build on NXP platforms OK. and board testing for imxrt/kinetis/lpc are OK

**Review by brycewilkins — APPROVED** [review]
Great to have this updated to the latest SDK and remove a lot of items from the patch list!

Also excellent idea to list the SHAs in the `README`.

**Comment by gmarull** [comment]
> > This PR adds yet more binary blobs (*.FLM), which makes things worse than they already are. Please, remove all the unnecessary stuff (which is a lot). This repo is cloned by everybody following the GSG, and so should be kept in decent shape.
> 
> @gmarull, the PR has been updated, kindly re-review.

I still see many files that should likely not be there:

image

Or... there's even a USB stack https://github.com/nxp-zephyr/hal_nxp/tree/mcux_2.15.000_RW61x/mcux/middleware/mcux-sdk-middleware-usb ? @jfischer-no

**Comment by mmahadevan108** [comment]
Thanks for catching this. I had missed deleting these `.ldt` files. We are working with the internal team to remove this file in the main `mcux-sdk` repo. 
https://github.com/nxp-zephyr/hal_nxp/tree/mcux_2.15.000_RW61x/mcux/middleware/mcux-sdk-middleware-usb is required for https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/usb/device/usb_dc_mcux.c USB driver.

**Comment by dleach02** [comment]
@mmahadevan108, can you rebase this with the current since we just merged the HWMv2 changes. Want to make sure things are still okay from a CI perspective.

**Comment by mmahadevan108** [comment]
@gmarull, can you take a look.

**Comment by mmahadevan108** [comment]
> @mmahadevan108, can you rebase this with the current since we just merged the HWMv2 changes. Want to make sure things are still okay from a CI perspective.
Rebased, @hakehuang . Could you do a test one more time to see if all looks fine with update to HWV2 and SDK 2.15 update.

**Review by danieldegrasse — APPROVED** [review]

**Pull request merged**